src/StdChains.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2 <0.9.0;
+
+pragma experimental ABIEncoderV2;
+
+import "./Vm.sol";
+
+abstract contract StdChains {
+    VmSafe private constant vm = VmSafe(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+    struct Chain {
+        // The chain name.
+        string name;
+        // The chain's Chain ID.
+        uint256 chainId;
+        // A default RPC endpoint for this chain.
+        // NOTE: This default RPC URL is included for convenience to facilitate quick tests and
+        // experimentation. Do not use this RPC URL for production test suites, CI, or other heavy
+        // usage as you will be throttled and this is a disservice to others who need this endpoint.
+        string rpcUrl;
+    }
+
+    bool private initialized;
+
+    // Maps from a chain's key (matching the alias in the `foundry.toml` file) to chain data.
+    mapping(string => Chain) private chains;
+    // Maps from a chain ID to that chain's key name
+    mapping(uint256 => string) private idToKey;
+
+    function getChain(string memory key) internal virtual returns (Chain memory) {
+        initialize();
+        return chains[key];
+    }
+
+    function getChain(uint256 chainId) internal virtual returns (Chain memory) {
+        initialize();
+        return chains[idToKey[chainId]];
+    }
+
+    function setChain(string memory key, string memory name, uint256 chainId, string memory rpcUrl) internal virtual {
+        chains[key] = Chain(name, chainId, rpcUrl);
+        idToKey[chainId] = key;
+    }
+
+    function initialize() private {
+        if (initialized) return;
+
+        setChain("anvil", "Localhost", 31337, "http://127.0.0.1:8545");
+        setChain("hardhat", "Localhost", 31337, "http://127.0.0.1:8545");
```

Comment:
Let's remove this since hardhat and anvil have the same chain ID, and users can overwrite the info if needed